cql3/query_processor.hh

```diff
@@ -143,7 +143,7 @@ public:
     static std::unique_ptr<statements::raw::parsed_statement> parse_statement(const std::string_view& query);
     static std::vector<std::unique_ptr<statements::raw::parsed_statement>> parse_statements(std::string_view queries);
 
-    query_processor(service::storage_proxy& proxy, service::forward_service& forwarder, data_dictionary::database db, service::migration_notifier& mn, service::migration_manager& mm, memory_config mcfg, cql_config& cql_cfg, utils::loading_cache_config auth_prep_cache_cfg, service::raft_group0_client& group0_client, std::shared_ptr<rust::Box<wasmtime::Engine>> wasm_engine);
+    query_processor(service::storage_proxy& proxy, service::forward_service& forwarder, data_dictionary::database db, service::migration_notifier& mn, service::migration_manager& mm, memory_config mcfg, cql_config& cql_cfg, utils::loading_cache_config auth_prep_cache_cfg, service::raft_group0_client& group0_client, std::shared_ptr<rust::Box<wasmtime::Engine>> wasm_engine, seastar::sharded<wasm::instance_cache>& wasm_instance_cache, std::shared_ptr<wasm::alien_thread_runner> alien_runner);
```

Comment:
The `rust::Box<wasmtime::Engine>> and `wasm::alien_thread_runner` seem to be needed by query processor only. Can they be wrappet together with a helper struct named `wasm_<something>`?

Reply:
I think that's a good idea, but I'm not sure how to name the wrapper. I'll try something in the rebase